dev/bots/analyze.dart

```diff
@@ -584,6 +587,44 @@ Future<void> verifyGoldenTags(String workingDirectory, { int minimumMatches = 20
   }
 }
 
+/// Use of Stopwatches can introduce test flakes as the logical time of a
+/// stopwatch can fall out of sync with the mocked time of FakeAsync in testing.
+/// The Clock object provides a safe stopwatch instead, which is paired with the
+/// FakeAsync as part of the test binding.
+final RegExp _findStopwatchPattern = RegExp(r'Stopwatch\(\)');
```

Comment:
@LongCatIsLooong was working on an analyzer-based way of detecting things like this instead of using somewhat error-prone string matching in https://github.com/flutter/flutter/pull/130101/files. This could be another great use case for that to exactly match on dart:core Stopwatch instantiations. Maybe we could extract the framework for these custom rules out of that PR and get the submitted without the more controversial debugAssert annotation so we could use it here as well?

Reply:
SG. I'll try converting this to using the analyzer package once that PR is ready.

Reply:
Awesome! Thank you @LongCatIsLooong!